prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTask.java

```diff
@@ -0,0 +1,171 @@
+		this.apiClient = apiClient;
+		this.api = api;
+		this.setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("api-group").type(WorkParameterType.TEXT).optional(false)
+				.description("API group of resource").build());
+		params.add(WorkParameter.builder().key("api-version").type(WorkParameterType.TEXT).optional(false)
+				.description("API version of resource").build());
+		params.add(WorkParameter.builder().key("kind-plural-name").type(WorkParameterType.TEXT).optional(false)
+				.description("Plural name of the resource kind. E.g. crontabs").build());
+		params.add(WorkParameter.builder().key("operation").type(WorkParameterType.TEXT).optional(false)
+				.description("Operation type/name. Can be one of " + Arrays.toString(OperationType.values())).build());
+		params.add(WorkParameter.builder().key("resource-json").type(WorkParameterType.TEXT).optional(true)
+				.description("The JSON to be used in create and update operations").build());
+		params.add(WorkParameter.builder().key("resource-name").type(WorkParameterType.TEXT).optional(true)
+				.description("Name of resource for get operation").build());
+		params.add(WorkParameter.builder().key("resource-namespace").type(WorkParameterType.TEXT).optional(true)
+				.description("Namespace of resource for get operation").build());
+		params.add(WorkParameter.builder().key("work-ctx-key").type(WorkParameterType.TEXT).optional(true)
+				.description("In get operation the result is stored in WorkContext with the provided key").build());
+		return params;
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskOutput> getWorkFlowTaskOutputs() {
+		return List.of(WorkFlowTaskOutput.OTHER);
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		try {
+			// Init kube client. We do not initialize it in c'tor because it is used as a
+			// Java bean.
+			// Therefore, failure to create client will fail the application.
+			initClient();
+
+			// Get the required parameters
+			String apiGroup = getRequiredParameterValue(workContext, "api-group");
+			String apiVersion = getRequiredParameterValue(workContext, "api-version");
+			String kindPluralName = getRequiredParameterValue(workContext, "kind-plural-name");
+			String operation = getRequiredParameterValue(workContext, "operation");
+
+			OperationType operationType = OperationType.valueOf(operation.toUpperCase());
+
+			switch (operationType) {
+				case UPDATE:
+					update(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case CREATE:
+					create(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case GET:
+					get(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+			}
+		}
+		catch (MissingParameterException | IOException | ApiException | IllegalArgumentException e) {
+			log.error("Kubeapi task failed", e);
```

Comment:
Add `operationType` to the log, to have more info

Reply:
if we add the operation type then why not add all extra details? what will you do with the operation alone?

Reply:
This function wrap Create, Update and Get operations, I think at least need to make it clear which operation (Maybe also need to consider even other parameters here)